jwst/flatfield/flat_field.py

```diff
@@ -405,8 +397,21 @@ def do_NIRSpec_flat_field(output_model,
         interpolated_flats = None
 
     any_updated = False
+    if exposure_type == "NRS_MSASPEC":
+        slits = nirspec.get_open_slits(output_model)
     for (k, slit) in enumerate(output_model.slits):
         log.debug("Processing slit %s", slit.name)
+        slit_nt = None
+        if exposure_type == "NRS_MSASPEC":
+            # Find this slit in the list of open slits.
+            for j in range(len(slits)):
+                if str(slits[j].name) == slit.name:
+                    slit_nt = slits[j]
+                    break
+            if slit_nt is None:
+                log.error("Couldn't find slit %s in list of open slits; "
```

Comment:
Is this actually the list of extracted slits. If so the message may be a bit misleading.